extras/module-node/test.c

```diff
@@ -0,0 +1,69 @@
+#include <stdio.h>
+#include "duktape.h"
+#include "duk_module_node.h"
+
+static duk_ret_t cb_resolve_module(duk_context *ctx) {
+	const char *module_id;
+	const char *parent_id;
+
+	module_id = duk_require_string(ctx, 0);
+	parent_id = duk_require_string(ctx, 1);
+
+	duk_push_sprintf(ctx, "%s.js", module_id);
+	printf("resolve_cb: id:'%s', parent-id:'%s', resolve-to:'%s'\n",
+		module_id, parent_id, duk_get_string(ctx, -1));
+
+	return 1;
+}
+
+static duk_ret_t cb_load_module(duk_context *ctx) {
+	const char *filename;
+	const char *module_id;
+
+	module_id = duk_require_string(ctx, 0);
+	duk_get_prop_string(ctx, 2, "filename");
+	filename = duk_require_string(ctx, -1);
+
+	printf("load_cb: id:'%s', filename:'%s'\n", module_id, filename);
+	duk_push_sprintf(ctx,
+		"if (module.id == 'pig.js') module.exports = 'you\\'re about to get eaten by %s';"
+		" else if (module.id == 'cow.js') module.exports = require('pig');"
+		" else if (module.id == 'ape.js') module.exports = { module: module, __filename: __filename, wasLoaded: module.loaded, size: 0 };",
+		module_id);
```

Comment:
That would probably be better, I agree.  This was kind of thrown together in 5 minutes, can you tell? :)